### ISSUE TITLE
Use MSVC vld1q_u8_ex and vst1q_u8_ex intrinsics for NEON code

### ISSUE THREAD

**Issue by ccawley2011** [zlib-ng/zlib-ng#1689]
MSVC provides variants of `vld1q_u8` and `vst1q_u8` with an additional parameter specifying the alignment in bits. It might be beneficial to make use of this where available.

**Comment by mtl1979** [comment]
As far as I know, those `_ex` variants only make difference for AArch32 and are equivalent to non-`_ex` versions on AArch64.

Unless we get alignment exceptions in the code, I don't see any benefit in using the `_ex` variants. We only support ARMv7 when using MSVC for 32-bit ARM.

**Comment by ccawley2011** [comment]
The [ARM documentation](https://developer.arm.com/documentation/den0018/a/NEON-Instruction-Set-Architecture/Alignment) states that on Cortex-A8 and Cortex-A9, a cycle is saved when higher alignments are specified, with the caveat that a bus error occurs if the alignment isn't met. For GCC and Clang, `__builtin_assume_aligned` can be used to provide similar functionality to the MSVC intrinsics.

I have a work in progress branch that makes use of this, but it's currently blocked by the issue mentioned here: https://github.com/zlib-ng/zlib-ng/issues/1842#issuecomment-2645962314

**Comment by nmoinvaz** [comment]
Solved by #1875 which has merged into `develop`.
